lib/teiserver/tachyon_battle.ex

```diff
@@ -54,6 +40,24 @@ defmodule Teiserver.TachyonBattle do
     end
   end
 
+  @spec start_battle(Teiserver.Autohost.id(), T.match_id()) ::
+          {:ok, T.id(), pid()} | {:error, term()}
```

Comment:
there should only be one spec per function, adding `when` for different clauses is ultimately the same function.